File: frontend/src/types/analysis.ts
import type { ISODateTime } from "./common";
import type { AIAdvisory } from "./advisory";
import type { AnomalyResult } from "./anomaly";
import type { DataQualityResult } from "./data-quality";
import type { LiquidityForecast } from "./forecast";
import type { RiskAssessment } from "./risk";

export interface AgentAnalysis {
  analysisId: string;
  agentId: string;
  dataQuality: DataQualityResult[];
  forecasts: LiquidityForecast[];
  anomalyResult: AnomalyResult;
  riskAssessment: RiskAssessment;
  advisory: AIAdvisory;
  alertId: string;
  caseId: string | null;
  calculatedAt: ISODateTime;
}

export interface RunAnalysisOptions {
  scenarioId?: string;
  forceRefresh?: boolean;
}

File: backend/app/schemas/analysis.py
from __future__ import annotations

from datetime import datetime

from app.schemas.advisory import AdvisorySummary
from app.schemas.anomaly import AnomalySummary
from app.schemas.common import AgentLensSchema
from app.schemas.data_quality import ProviderDataHealth
from app.schemas.forecast import ForecastSummary
from app.schemas.risk import RiskAssessment


class AnalysisSummary(AgentLensSchema):
    id: str
    created_at: datetime
    data_quality: list[ProviderDataHealth]
    forecasts: list[ForecastSummary]
    anomaly: AnomalySummary
    risk: RiskAssessment
    advisory: AdvisorySummary
